peachjam/middleware.py

```diff
@@ -48,6 +51,15 @@ def process_request(self, request):
             del request.META["HTTP_ACCEPT_LANGUAGE"]
 
 
+class SetPreferredLanguageMiddleware(MiddlewareMixin):
+    def process_response(self, request, response):
+
+        if hasattr(request, "set_language") and "test" not in sys.argv:
+            request.POST = {"language": request.set_language}
+            response = set_language(request)
```

Comment:
The only important thing that `set_language` does is set the cookie. We have a response object here which we can add a cookie to, without going through a weird redirect.

Why not just replace this code with:

```python
            response.set_cookie(
                settings.LANGUAGE_COOKIE_NAME,
                lang_code,
                max_age=settings.LANGUAGE_COOKIE_AGE,
                path=settings.LANGUAGE_COOKIE_PATH,
                domain=settings.LANGUAGE_COOKIE_DOMAIN,
                secure=settings.LANGUAGE_COOKIE_SECURE,
                httponly=settings.LANGUAGE_COOKIE_HTTPONLY,
                samesite=settings.LANGUAGE_COOKIE_SAMESITE,
            )
```